### PR TITLE
Card no #5

### PR DESCRIPTION
**Pull request by sayalis-cuelogic**
Implemented View Post Feature.
As per previous Pull Request comment implemented User Auth Model.
Used request.user.

### CODE REVIEW

File: blogauth/views.py [hunk 1]
```diff
@@ -57,19 +60,23 @@ def get_queryset(self):
...
 
     def post(self, request):
-        form = RegForm(request.POST)
         data = request.POST
-        username = data.get('username')
-        password = data.get('password1')
-        user = User.objects.get(username=username, password=password)
-        if (user):
-            return render(request, 'post/blogpage.html', {'user': user})
+        form = LoginForm(request.POST)
```
Comment (aloka-cuelogic) [inline blogauth/views.py]: This Custom Authentication is not required this is by default provided by django and you need to call the same.

For defining page after successful login use settings.py and define LOGIN_REDIRECT_URL

**Comment by aloka-cuelogic** [comment]
@sayalis-cuelogic 

Please use login_required decorator as add blog page and other's page are directly opening even without user getting logged in.

**Comment by sayalis-cuelogic** [comment]
Worked on review comments.
Used login provided by django

**Comment by sayalis-cuelogic** [comment]
Used login_required decorator for add and view blog page.

File: post/views.py [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
 
-from blogauth.forms import RegForm
 from post.forms import PostForm
 from post.models import PostInfo
 
 
 class BlogPageView(FormView):
     template_name = 'post/blogpage.html'
-    form_class = RegForm
+    login_required = True
```
Comment (aloka-cuelogic) [inline post/views.py]: remove this from the code as you have moved this in urls.py.

File: post/views.py [hunk 3]
```diff
@@ -28,23 +29,25 @@ def dispatch(self, *args, **kwargs):
 class PostPageView(FormView):
     template_name = 'post/blogadd.html'
     form_class = PostForm
+    login_required = True
```
Comment (aloka-cuelogic) [inline post/views.py]: same as above comment

**Comment by sayalis-cuelogic** [comment]
Removed get_queryset from EmailAvailabilityView

**Comment by aloka-cuelogic** [comment]
@sayalis-cuelogic 

Working as expected, also used django auth.
Merging PR.

**Comment by aloka-cuelogic** [comment]
@sayalis-cuelogic 

From Next task on-wards try to avoid using @csrf_exempt and send the actual csrf token?
How to do same please look into following:

https://github.com/CuelogicTech/prepairit/blob/dev/static/js/account_scripts.js#L46

**Pull request merged**